src/js/hooks/cycleCount/useCountStepTable.jsx

```diff
@@ -195,6 +183,37 @@ const useCountStepTable = ({
         if (rowIndex !== null && columnId && error !== null) {
           setError(null);
         }
+        const handleClick = (event) => {
+          if (rowIndex === null && columnId === null) {
+            return;
+          }
+
+          const { target } = event;
+          const isInputElement = target.closest('input, select, textarea, .date-field-input, .react-datepicker, .react-select__control');
+          const isDatePickerDay = target.closest('.react-datepicker__day');
+          const isSelectOption = target.closest('.react-select__option');
```

Comment:
please add comments explaining what this code is responsible for - how does it fix performance issues etc.
Additionally, the naming doesn't feel quite right here:

- `isSelectOption` - is selection option what?
- `isDatePickerDay` - what exactly? is it present? is it picked? is it empty?